### DISCUSSION TITLE
Total Prefixes generated by lspgen

### DISCUSSION THREAD

**Discussion by DivyaBharathiS** [rtbrick/bngblaster#111] [Q&A] [answered]
Hello,

I have a query on the total number of prefixes generated by lspgen.

Topology Used:  Bngblaster --------- Router1

**example 1. Node count: 3000 and External Prefix Count: 30000**

lspgen --connector 1921.6100.1003 --no-ipv6 --external-count 30000 --node-count 3000 --sequence 5 --mrt-file isis.mrt
Oct 03 10:08:44.593600 Add context for instance default, protocol isis, topology unicast
Oct 03 10:08:44.593691 Add connector to 0x192161001003
Oct 03 10:08:44.593701 LSP generation parameters
Oct 03 10:08:44.593708  Area 49.0001/24
Oct 03 10:08:44.593713  Level 1, sequence 0x5, lsp-lifetime 65535
**Oct 03 10:08:44.593724  IPv4 Node Base Prefix 192.168.0.0/32, End Prefix 192.168.11.183/32, 3000 prefixes
Oct 03 10:08:44.593730  IPv4 Link Base Prefix 172.16.0.0/31, End Prefix 172.16.46.222/31, 6000 prefixes
Oct 03 10:08:44.593735  IPv4 External Base Prefix 10.0.0.0/28, End Prefix 10.7.82.240/28, 30000 prefixes**
Oct 03 10:08:44.593741  SRGB base 10000, range 3000
Oct 03 10:08:44.593749 Generating a graph of 3000 nodes and 6000 links
Oct 03 10:08:47.714679  Root node 1921.6800.0000.00

From lspgen, the total prefixes are 30000 external prefixes + 3000 node prefixes + 6000 link prefixes (Total: 39000).I am expecting 39K prefixes to be received and downloaded on Router1. But the prefix count is 45007 prefixes on Router1 (that includes R1's 5 isis routes).

Could you clarify on how did we arrive at 45K prefixes for the above mentioned example ?

Thanks

**Comment by hannesgredler** [comment, marked as answer]
Hi Divya - Your math is correct - There should be 39K prefixes.
This is fixed in dev branch -> https://github.com/rtbrick/bngblaster/commit/2c2704c7d83dbf0a0416ed1fcca869747590cfb6

**Reply by DivyaBharathiS** [reply]
Thank you Hannes

**Comment by GIC-de** [comment]
Fixed in 0.7.13!
